actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -59,7 +59,7 @@ def check_path_parameters!
       path_parameters.each do |key, value|
         next unless value.respond_to?(:valid_encoding?)
         unless value.valid_encoding?
-          raise ActionController::BadRequest, "Invalid parameter: #{key} => #{value}"
```

Comment:
There's definitely *something* to be fixed; this line clearly thinks it's specifying a message, which is not actually the case.

And yes, as the bot pointed out, we should be working against master -- any necessary backporting will be handled after that.

As for the choice of how we actually solve the problem, I'm not sure... I'm hesitant to go with the code as proposed, partly because of the compatibility change you've already mentioned, but also because this addresses the "exception can't be used 'normally' because it has magic / non-message parameters" problem by introducing a new exception class which *also* has magic / non-message parameters.

Finally, I wonder whether we should really be revisiting the whole `original_exception` arrangement, everywhere it's used, given that we can now rely on `cause`.